dist/tools/jlink/jlink.sh

```diff
@@ -58,7 +58,12 @@ _JLINK_SERVER=JLinkGDBServer
 _JLINK_IF=SWD
 _JLINK_SPEED=2000
 # default starting address of the devices flash memory
-_FLASH_ADDR=0
+if [ -n "${FLASH_OFFSET}" ]; then
+    _FLASH_ADDR=${FLASH_OFFSET}
+    echo "Flashing at offset ${FLASH_OFFSET}"
+else
+    _FLASH_ADDR=0
+fi
```

Comment:
> if this var is really needed by the softdevice (and I think it is rather the softdevice adaption code that needs it...), it might as well be defined only by the softdevice specific code with a simple

The softdevice binary import instructions modifies this variable (which is **only** used by jlink.sh) to flash the RIOT code in a specific part of the ROM, but there's no other code in RIOT which needs to change this variable or makes use of it.

> I disagree: having this 'standard' global variable does not hurt in any case. If the flash tool does not need it, simply don't use it. If there is something (whatever that my be) that needs to know the starting address of the flash, we have a defined way to define/retrieve it.

An example of what it needs to be done for other scripts is given in #7181, specifically here: 2aa95a7841b56755c84b460eabd94a10dee4f992 where I do the modifications needed to use a "global" variable. Of course I agree to use a "standard" (if we can call it like that) global variable, I was only talking about the modifications needed to remove the current one to replace it by this "global" one (namely change the Makefile for the softdevice, very simple) which, IMHO, is out of the scope of this PR.

Anyways, if it's ok for you to do that change in this PR I can of course do it.